### PR TITLE
[NOSQUASH] Remove the unnecessary sort in CNullDriver::addTexture

### PR DESCRIPTION
**Pull request by Desour**
TL;DR: A performance fix for faster client startup.

I've done some little profiling on client connect. Here's a flame graph showing Game::startup of a session where it took about 40 seconds to connect to a server:

graph_long_before

The `std::__introsort_loop` is the sort of CNullDriver::Textures that we do for each texture. This sorting is unnecessary.

CNullDriver::Textures holds an irr::core::array that is possibly sorted by texture name. Textures are accessed via binary search (see irrArray.h), which always sorts anyway. The comment that was already there also explains this.
CNullDriver::Textures is also not used by any child class, FYI.

Benchmarks (of Client::afterContentReceived, and Game::startup in parenthesis):
(I haven't run these often. Values are rounded.)

lightweight scenario (singleplayer mtg + mesecons):
before: 1.1 s (2.0 s)
after: 0.9 (1.8)

heavy scenario (some server):
before: 35.9 s (44.4 s)
after: 5.0 s (12.7 s)

### CODE REVIEW

**Comment by sfan5** [comment]
I don't think we use this but this sort call can be removed too:
grafik

**Comment by Desour** [comment]
`renameTexture` is actually broken, if I'm not wrong, because `Textures` is not flagged as unsorted before the `Textures.sort()` call.
I've removed the function now, together with another unused not useful one.

**Comment by appgurueu** [comment]
(For the future) could this be refactored to use an unordered map?

**Comment by sfan5** [comment]
As I just found out [here] texture names are apparently not an one-to-one mapping to textures :thinking: 
So, maybe.


[here]: https://irc.minetest.net/minetest-dev/2023-09-13#i_6113112

**Comment by Desour** [comment]
If they are not supposed to be unique, we could still use an unordered multimap.
It would make sense imo, I just didn't do it because I wanted the change to be simple. (Also, if I change the header, I have to recompile half of minetest.)

**Comment by SmallJoker** [comment]
The Irrlicht situation, visualized:
xkcd Standards

 @BeverlyCode Please feel free to forward such patches to upstream Irrlicht so that all forks *could* benefit from it. Fixes in private forks do not help anyone else.

**Comment by appgurueu** [comment]
> @BeverlyCode Please feel free to forward such patches to upstream Irrlicht so that all forks _could_ benefit from it. Fixes in private forks do not help anyone else.

(Or just publicize your currently "private" fork so we can take a look).

Also feel free to responsibly disclose any vulnerabilities to us and/or Irrlicht as applicable.

**Comment by appgurueu** [comment]
I looked through Irrlicht's commit history a bit, and it seemed to me like any recent security improvements (in loaders) stemmed from IrrlichtMT. Could you point to a specific commit, or as said, disclose the vulnerability to us? Are you sure there is presently a vulnerability in IrrlichtMT?

**Comment by sfan5** [comment]
Our primary concern are security-relevant issues in the engine or IrrlichtMt.
You can disclose them as written in our [security policy](https://github.com/minetest/minetest/security/policy) and they will absolutely get looked at.

> and to be honest I don't really like any of you. 🤷

That is regrettable but it was your choice to comment here.
